### PR TITLE
fix: Change wording in docs from "builder methods" to "fluent setters"

### LINKED ISSUE #650: Change wording in docs from "builder methods" to "fluent setters"
> Actually these should be called "fluent setters" not "builder methods". Builder methods usually refer to a separate builder object that may have incomplete fields and need to be consumed to become the built object, e.g. https://docs.rs/typed-builder.

_Originally posted by @SOF3 in https://github.com/ratatui-org/ratatui/issues/647#issuecomment-1827029219_

---

As part of making sure we use consistent and accurate language in docs, use "setter methods" / "fluent setters" instead of "Builder methods" or "builders" to refer to the methods in Ratatui types that set values.

### PR DESCRIPTION
**Pull request by TieWay59**
Fixes #650

### CODE REVIEW

**Review by orhun — APPROVED** [review]
Definitely not a word that I'm used to but +1 for doing things by the book.

**Review by joshka — CHANGES_REQUESTED** [review]

File: src/layout.rs [hunk 1]
```diff
@@ -265,7 +265,7 @@ impl Layout {
             .is_ok()
     }
 
-    /// Builder method to set the constraints of the layout.
+    /// Fluent setters to set the constraints of the layout.
```
[the remark this answers is not in the document]
Comment (TieWay59) [reply src/layout.rs:268]: Sorry for my late reply. I prefer this suggestion. For a user, simple expressions can help read. We can fix the old wrong version and just use `set` for clearness.

> combine this with adding #[must_use] attributes on all the setters?

I'm not sure what to do about this. Do you mean check all setters and give the  `#[must_use]` & `set doc`?

[the remark this answers is not in the document]
Comment (joshka) [reply src/layout.rs:268]: I think any setter method that takes self and returns self needs to have a #[must_use] attribute. This indicates that the caller must do something with the return value and makes it clear that it's an error instead of a bug to write code that doesn't use the return value like below. Because this changes the same code (i.e. every setter method in Ratatui), it's worth bundling this change into a single large change where we add the same language across the entire crate and also add that attribute so we don't miss any of the places where this is necessary. Alternatively just fix the comments here and we can split that out to two more tasks (add consistent comments for all setters and add must_use on all fluent setters).

```rust
let layout = Layout::default();
layout.constraints([Min(0)...]);
```

[the remark this answers is not in the document]
Comment (TieWay59) [reply src/layout.rs:268]: I agree with "less is more". 

Let's not focus on the term for now. `not modifying the underlying object` is not the direct reason it must be used. 

When users invoke a setter, they tend to make some changes on the initial instance. I prefer to explain that:

 = note: `The setters have moved & returned the original instance, so the yield should be used or this operation is redundant.`

@joshka Do you think this wording fits your requirements? I think this is clear enough for new devs. `modifying` is the point I didn't get in the first place.

We can still express fluent setters in the code docs, I think it's ok to be formal there.

[the remark this answers is not in the document]
Comment (joshka) [reply src/layout.rs:268]: That seems a bit wordy / repeats the existing not about using the value. What about:
"method moves the value of self and returns the modified value"?

[the remark this answers is not in the document]
Comment (TieWay59) [reply src/layout.rs:268]: Great! Very straightforward. It's so nice to discuss with you. I'll follow this on my next commit.

**Comment by TieWay59** [comment]
@joshka  I made a WIP commit, do you think this is correct? I'll follow this pattern if this sample is fine.

**Comment by joshka** [comment]
> @joshka I made a WIP commit, do you think this is correct? I'll follow this pattern if this sample is fine.

LGTM - go for it

**Comment by joshka** [comment]
Oh BTW if you want to find all the places that could be must_use like this automatically - there's a clippy lint that you can use. It probably shouldn't be enabled by default as I suspect it would probably have some noise (things we don't want to make must_use), but it might help.

```shell
cargo clippy -- -W clippy::must-use-candidate
```

**Comment by joshka** [comment]
Sounds good

File: src/widgets.rs [hunk 2]
```diff
@@ -1,6 +1,6 @@
 //! `widgets` is a collection of types that implement [`Widget`] or [`StatefulWidget`] or both.
 //!
-//! All widgets are implemented using the builder pattern and are consumable objects. They are not
+//! All widgets are implemented by the fluent interface and are consumable objects. They are not
```
Comment (joshka) [inline src/widgets.rs:3]: I think we can drop this wording entirely - it's not particularly relevant.

```suggestion
//! Widgets are created for each frame as they are consumed by when rendered. They are not
```

Perhaps as a second sentence "Most widgets implement their fluent style setter methods like `fn(mut self, value: ...) -> Self`. The results of each setter call must be chained with another call or used.

You'll probably need to reformat the code locally if you take this suggestion (so don't just merge it from github)

File: src/widgets/sparkline.rs [hunk 3]
```diff
@@ -18,7 +18,7 @@ use crate::{
 /// `Sparkline` can be styled either using [`Sparkline::style`] or preferably using the methods
 /// provided by the [`Stylize`](crate::style::Stylize) trait.
 ///
-/// # Builder methods
+/// # Fluent setters
```
Comment (joshka) [inline src/widgets/sparkline.rs:21]:
```suggestion
/// # Setter methods
```

**Review by joshka — CHANGES_REQUESTED** [review]
LGTM generally - a couple of small things

**Comment by TieWay59** [comment]
@joshka  Ok, I've fixed them now.

**Review by joshka — APPROVED** [review]

**Comment by joshka** [comment]
Thanks for the small fixes. Would you mind please squashing your commits locally (so that that last commit is signed and I can merge it)?

**Comment by TieWay59** [comment]
> Thanks for the small fixes. Would you mind please squashing your commits locally (so that that last commit is signed and I can merge it)?

@joshka Sure, I've squashed it.

**Review by joshka — APPROVED** [review]

**Comment by TieWay59** [comment]
> Hrm - the commit is still not gpg signed (so I get a merging is blocked error)

@joshka Sorry, I was at my workplace without my PC. It took me some time to go back and reach my own PC.
